woocommerce/woocommerce
# Add filters to prevent quantity adjustment/removal notices in checkout blocks

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This adds a filter to allow extensions to prevent notices from being shown in scenarios where they don't make sense.

Related to https://github.com/woocommerce/woocommerce-composite-products/issues/1140.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

Install the following plugin:
[test-filters-plugin.zip](https://github.com/user-attachments/files/17872796/test-filters-plugin.zip)

1. Create a composite product.
2. Add the product to the cart.
3. Go to the Cart block and alter the quantity of the main product.
4. Verify that no notices are shown regarding quantity adjustment of the children products. 

<img width="1462" alt="image" src="https://github.com/user-attachments/assets/4eeddbe0-0c17-4452-8a69-6944f9f1754d">

> These should NOT show up


5. Remove the main product from the cart.
6. Verify that no notices are shown regarding the removal of the children products.

<img width="1397" alt="image" src="https://github.com/user-attachments/assets/6d1af74e-997b-4461-84f0-2f6a513c969e">

> These should NOT show up



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Add filters to prevent quantity adjustment or removal notices in checkout blocks</summary>

#### Comment 

</details>
